src/ipc/ipc-common/IPCProxyModelProperty.h

```diff
@@ -139,16 +135,19 @@ class IPCProxyModelProperty : public facelift::ModelProperty<ModelDataType>
                     --last;
                 }
 
-                QList<ModelDataType> list;
-                m_proxy.ipc()->sendMethodCallWithReturn(requestMemberID, list, first, last);
+                std::tuple<int, QList<ModelDataType>> requestResult;
+                m_proxy.ipc()->sendMethodCallWithReturn(requestMemberID, requestResult, first, last);
 
-                Q_ASSERT(list.size() == (last - first + 1));
+                first = std::get<0>(requestResult);
+                auto &list = std::get<1>(requestResult);
+                // Check if the server returned the requested item
```

Comment:
you are checking if the list is empty on the line below !

Reply:
done - deleted comment at all.